File: py/1709.py
def main():
    half = int(input())
    half //= 2
    count, start, position = 1, 1, 1

    while position != 0:
        if start < half:
            position = start
            start = 2 * position + 1

        else:
            position = start - half
            start = 2 * position

        count += 1

    print(count)


main()

File: cpp/1709.cpp
#include <iostream>

using namespace std;

int main()
{
    int half, count = 1, start = 1, position = 1;
    cin >> half;
    half /= 2;

    while (position != 0) {
        if (start < half) {
            position = start;
            start = 2 * position + 1;
        } else {
            position = start - half;
            start = 2 * position;
        }

        count++;
    }

    cout << count << '\n';
    return 0;
}